init.sls

```diff
@@ -12,6 +12,19 @@ gitlab_repo:
 {{ edition }}:
   pkg.installed: []
 
+# Deploy signing_key.gpg if options are given in pillar.
+# Make sure it's not existent if not.
+{% if salt['pillar.get']('gitlab:gitaly:signing_key', None) != none %}
```

Comment:
As `/etc/gitaly` is not present in the default install, maybe we can place the keys within `/etc/gitlab`? 

Also, I think we should support SSH keys. We could also infer the filename from pillars, something like this

```yaml
gitaly:
  signing_key:
    gitaly_ui.ssh: 'ssh-ed25519 AAAA...'

    # Alternatively
    gitaly_ui.gpg: |
       ---- BEGIN -----
```

I'm not sure if the file extention is somehow valued, if it's inferred from the content we could just use `/etc/gitlab/gitaly-ui.key` instead.

Reply:
SSH keys should be supported with the proposed solution already, as you can only provide one signing_key and that signing_key will then just be used by gitaly. If you want to use a ssh key, you just have to provide it as `gitlab.gitaly.signing_key`

Reply:
Regarding changing the path: Sounds good :+1: